text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,895 @@
+              schema:
+                $ref: '#/components/schemas/GetOutputsByAddressResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/post-outputs-by-address-response-example-1'
+        "400":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        "404":
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+
+components:
+
+  examples:
+    invalid-data-provided-response-example:
+      value:
+        error:
+          code: 400
+          message: "invalid data provided"
+    can-not-find-data-response-example:
+      value:
+        error:
+          code: 404
+          message: "can not find data"
+    get-info-response-example-1:
+      value:
+        data:
+          name: Hornet
+          version: 0.5.2
+          isHealthy: true
+          operatingNetwork: Mainnet
+          peers: 5
+          coordinatorAddress: UDYXTZBE9GZGPM9SSQV9LTZNDLJIZMPUVVXYXFYVBLIEUHLSEWFTKZZLXYRHHWVQV9MNNX9KZC9D9UZWZ
+          isSynced: true
+          latestMilestoneHash: JXVC9LGIEPCEJLEN9EXOKGUBFXOZDTZYIMZMPIJCGUALBENVTTRFYVUCKZCOVPRZKEUYZGYPSQGAA9999
+          latestMilestoneIndex: 1699556
+          latestSolidMilestoneHash: JXVC9LGIEPCEJLEN9EXOKGUBFXOZDTZYIMZMPIJCGUALBENVTTRFYVUCKZCOVPRZKEUYZGYPSQGAA9999
+          latestSolidMilestoneIndex: 1699556
+          pruningIndex: 1696326
+          time: 1599584382
+          features:
+            - Plugin X
+            - Plugin Y
+    get-tips-response-example-1:
+      value:
+        data:
+          tip1: "f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d"
+          tip2: "78d546b46aec4557872139a48f66bc567687e8413578a14323548732358914a2"
+    get-tips-response-example-2:
+      value:
+        data:
+          tip1: "f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d"
+          tip2: "f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d"
+    get-messages-by-hash-response-example-1:
+      value:
+        data:
+          version: 1
+          parent1: "f532a53545103276b46876c473846d98648ee418468bce76df4868648dd73e5d"
+          parent2: "78d546b46aec4557872139a48f66bc567687e8413578a14323548732358914a2"
+          payloadLength: 0
+          payload:
+            type: 2
+            index: "5350414d"
+            data: "SGVsbG8gd29ybGQh"
+    post-messages-by-hash-response-example-1:
+      value:
+        data:
+          "MESSAGE HASH X":
```

Comment:
I think `data` should be an array of messages and not an object containing messages by their hash. The hash should be integrated into the message object within the array.

Reply:
How would you do it in `GET /messages-by-tag` for example? Would you do something like 

```json
{
  "data": [
    {
      "abc": [
        {
          "hash": "1c6943b0487c92fd057d4d22ad844cc37ee27fe6fbe88e5ff0d20b2233f75b9d",
          "version": 1,
          "parent1": "ae07bf7ee5117a8fac99cb9969cfec17e74b3d5a575171e172d8a6e5bcbb237c",
          "parent2": "d33ec3e5a9e4d5759923f10771ce61d62627ec12464fc600dd46d58c542e5fd2",
          "payload": {
            "type": 2,
            "index": "abc",
            "data": "SGVsbG8gd29ybGQh"
          }
        }
      ]
    }
  ]
}
```

or would you do something like:

```json
{
  "data": [
    {
      "tag": "abc",
      "hash": "1c6943b0487c92fd057d4d22ad844cc37ee27fe6fbe88e5ff0d20b2233f75b9d",
      "version": 1,
      "parent1": "ae07bf7ee5117a8fac99cb9969cfec17e74b3d5a575171e172d8a6e5bcbb237c",
      "parent2": "d33ec3e5a9e4d5759923f10771ce61d62627ec12464fc600dd46d58c542e5fd2",
      "payload": {
        "type": 2,
        "index": "abc",
        "data": "SGVsbG8gd29ybGQh"
      }
    }
  ]
}
```

Reply:
The first one, per tag, multiple messages can be in the array so the 2nd doesn't even work.